imports/server/hunts.ts

```diff
@@ -74,7 +79,90 @@ function renderExistingJoinEmail(setting: SettingType | undefined, user: Meteor.
   return Mustache.render(DEFAULT_EXISTING_JOIN_TEMPLATE, view);
 }
 
+const SavedDiscordObjectFields = {
+  id: String,
+  name: String,
+};
+
+const HuntShape = {
+  name: String,
+  mailingLists: [String],
+  signupMessage: Match.Optional(String),
+  openSignups: Boolean,
+  hasGuessQueue: Boolean,
+  submitTemplate: Match.Optional(String),
+  homepageUrl: Match.Optional(String),
+  puzzleHooksDiscordChannel: Match.Optional(SavedDiscordObjectFields),
+  firehoseDiscordChannel: Match.Optional(SavedDiscordObjectFields),
+  memberDiscordRole: Match.Optional(SavedDiscordObjectFields),
+};
+
 Meteor.methods({
+  createHunt(value: unknown) {
+    check(this.userId, String);
+    checkAdmin(this.userId);
+    // Despite looking right to me, TS seems to trip over the [String] pattern for mailingLists.
+    // @ts-ignore Something is wrong with check's array-of-strings typing
+    check(value, HuntShape);
+
+    const huntId = Hunts.insert(value as HuntInsertFields);
+
+    // Sync discord roles
+    MeteorUsers.find({ hunts: huntId })
+      .forEach((u) => {
+        addUserToDiscordRole(u._id, huntId);
+      });
+
+    return huntId;
+  },
+
+  updateHunt(huntId: unknown, value: unknown) {
+    check(this.userId, String);
+    checkAdmin(this.userId);
+    check(huntId, String);
+    // Despite looking right to me, TS seems to trip over the [String] pattern for mailingLists.
+    // @ts-ignore Something is wrong with check's array-of-strings typing
```

Comment:
Feel free to take a crack at winning a duel with the typechecker here; I gave up and moved on.

(In an ideal world, maybe we'd be able to synthesize these match patterns from the io-ts types or something?  Or use a `Match.Where` that calls the codec to see if it deserializes cleanly?  But maybe we'd rather find some other approach with smaller codesize?  🤷🏻‍♂️ )

Reply:
It seems like the issue is that Typescript is inferring the type of `[String]` to be an array of unspecified size (`StringConstructor[]`), rather than a tuple type (`[StringConstructor]`), and that resolution happens at assignment time and isn't updated or refined when the object is passed into `check`.

Reply:
Also we could use `Match.Where` but when I originally added the typing I didn't make the typing for that specific enough. Right now it's

```
        function Where(condition: (val: any) => boolean): Matcher<any>;
```

and I think it would need to be something like

```
        function Where<T>(condition: (val: any) => val is T): Matcher<T>;
```

Reply:
Delayed update: I believe that once DefinitelyTyped/DefinitelyTyped#55140 lands and we pick it up, we could replace this with `check(value, Match.Where(HuntCodec.is))`.